draft-ietf-webtrans-http3.md

```diff
@@ -366,10 +366,45 @@ an implementation to choose what stream or datagram to discard.
 
 # Session Termination
 
-An WebTransport session over HTTP/3 is terminated when either endpoint closes
-the stream associated with the CONNECT request that initiated the session.
-Upon learning about the session being terminated, the endpoint MUST stop
-sending new datagrams and reset all of the streams associated with the session.
+A WebTransport session over HTTP/3 is considered terminated when either
+endpoint closes the stream associated with the CONNECT request that initiated
+the session.  Upon learning about the session being terminated, the endpoint
+MUST stop sending new datagrams and reset all of the streams associated with
+the session.
+
+To terminate a session with a detailed error message, an application MAY send
+an HTTP capsule [HTTP3-DATAGRAM] of type CLOSE_WEBTRANSPORT_SESSION (0x2843).
+The format of the capsule SHALL be as follows:
+
+~~~
+CLOSE_WEBTRANSPORT_SESSION Capsule {
+  Application Error Code (32),
+  Application Error Message (..8192),
+}
+~~~
+
+CLOSE_WEBTRANSPORT_SESSION has the following fields:
+
+Application Error Code:
+
+: A 32-bit error code provided by the application closing the connection.
+
+Application Error Message:
+
+: A UTF-8 encoded error message string provided by the application closing the
+  connection.  The message takes up the remainer of the capsule, and its
+  length MUST NOT exceed 1024 bytes.
```

Comment:
Any particular reason for the choice of length? Just trying to stay under QUIC minimum sizes?

Reply:
Not in particular, just trying to place some bound on the string length, and 1024 seemed like a reasonable number.

Reply:
Maybe add a sentence explaining the limit?